Android/Tapp/app/src/main/java/app/anish/com/tapp/dialogs/LinkedInWebViewLoginDialog.java

```diff
@@ -0,0 +1,294 @@
+    private static final String AUTHORIZATION_URL = "https://www.linkedin.com/uas/oauth2/authorization";
+    private static final String ACCESS_TOKEN_URL = "https://www.linkedin.com/uas/oauth2/accessToken";
+    private static final String SECRET_KEY_PARAM = "client_secret";
+    private static final String RESPONSE_TYPE_PARAM = "response_type";
+    private static final String GRANT_TYPE_PARAM = "grant_type";
+    private static final String GRANT_TYPE = "authorization_code";
+    private static final String RESPONSE_TYPE_VALUE ="code";
+    private static final String CLIENT_ID_PARAM = "client_id";
+    private static final String STATE_PARAM = "state";
+    private static final String REDIRECT_URI_PARAM = "redirect_uri";
+
+
+    private static final String QUESTION_MARK = "?";
+    private static final String AMPERSAND = "&";
+    private static final String EQUALS = "=";
+
+    private static final String TOKEN_VALUE_JSON_KEY = "access_token";
+    private static final String TOKEN_EXPIRE_JSON_KEY = "expires_in";
+
+
+    private ProgressDialog mProgressDialog;
+    private AuthListener authListener;
+    private WebView mWebView;
+
+    public LinkedInWebViewLoginDialog(Context context, AuthListener authListener) {
+        super(context);
+        this.authListener = authListener;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        requestWindowFeature(Window.FEATURE_NO_TITLE);
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.webview_layout);
+
+        initWebView();
+        loadAuthUrl();
+    }
+
+    private void initWebView() {
+        mWebView = (WebView) findViewById(R.id.wvWebView);
+        mWebView.requestFocus(View.FOCUS_DOWN);
+        mWebView.setWebViewClient(new LinkedinWebViewClient());
+    }
+
+    private void loadAuthUrl() {
+        String authUrl = getAuthorizationUrl();
+        mProgressDialog = ProgressDialog.show(getContext(), "", getContext().getString(R.string.loading),true);
+        mWebView.loadUrl(authUrl);
+    }
+
+
+    /**
+     * Method that generates the url for get the access token from the Service
+     * @return Url
+     */
+    private static String getAccessTokenUrl(String authorizationToken) {
+        return TextUtils.join("",
+                new String [] {
+                        ACCESS_TOKEN_URL,QUESTION_MARK,GRANT_TYPE_PARAM,
+                        EQUALS, GRANT_TYPE, AMPERSAND, RESPONSE_TYPE_VALUE,
+                        EQUALS, authorizationToken, AMPERSAND, CLIENT_ID_PARAM,
+                        EQUALS, LINKEDIN_API_KEY, AMPERSAND, REDIRECT_URI_PARAM,
+                        EQUALS, REDIRECT_URI, AMPERSAND, SECRET_KEY_PARAM,EQUALS,
+                        LINKEDIN_SECRET_KEY
+                }
+                );
+    }
+
+    /**
+     * Method that generates the url for get the authorization token from the Service
+     * @return Url
+     */
+    private static String getAuthorizationUrl() {
```

Comment:
This is similar to my previous comment. You've done a lot of work here that you didn't need to do if you applied the libraries I mentioned.